spec/test_data/v4.2/expected_spell_book.rbi

```diff
@@ -72,73 +72,76 @@ module SpellBook::ModelRelationShared
   sig { returns(SpellBook::Relation) }
   def all(); end
 
+  sig { params(block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
+  def unscoped(&block); end
+
   sig { params(args: T.untyped).returns(SpellBook::Relation) }
   def recent(*args); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def select(*args, block); end
+  def select(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def order(*args, block); end
+  def order(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def reorder(*args, block); end
+  def reorder(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def group(*args, block); end
+  def group(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def limit(*args, block); end
+  def limit(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def offset(*args, block); end
+  def offset(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def joins(*args, block); end
+  def joins(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def where(*args, block); end
+  def where(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def rewhere(*args, block); end
+  def rewhere(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def preload(*args, block); end
+  def preload(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def eager_load(*args, block); end
+  def eager_load(*args, &block); end
 
   sig { params(args: T.untyped, block: T.nilable(T.proc.void)).returns(SpellBook::Relation) }
-  def includes(*args, block); end
+  def includes(*args, &block); end
```

Comment:
This seems to work on [sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Aclass%20Site%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20sig%20%7B%20params(a%3A%20T.untyped%2C%20block%3A%20T.nilable(T.proc.void)).returns(String)%20%7D%0A%20%20def%20self.where(*a%2C%20block)%0A%20%20%20%20%22String%22%0A%20%20end%0Aend%0A%0AT.reveal_type(Site.where(id%3A%201))%0AT.reveal_type(Site.where(id%3A%201)%20%7B%20'23'%20%7D%20))
```
./sorbet-test.rb:4: Revealed type: Site::Relation https://srb.help/7014
     4 |T.reveal_type(Site.includes(:id, :name))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  From:
    ./sorbet-test.rb:4:
     4 |T.reveal_type(Site.includes(:id, :name))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
```

But not when it's a standalone function on [sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7B%20params(block%3A%20T.nilable(T.proc.void)).returns(String)%20%7D%0Adef%20with_block(block)%0A%20%20'asdf'%0Aend%0A%0AT.reveal_type(with_block())%0AT.reveal_type(with_block%20%7B%20'asd'%20%7D))

It looks like a bug. I'll report it to the sorbet team.

Reply:
^ `block` actually works when it's a class method.
However, I still think the fix to use `&block` is better & more consistent.